ADALiOS/ADALiOS/ADAuthenticationContext+WebRequest.m

```diff
@@ -295,9 +296,9 @@ - (NSString*)queryStringFromResource:(NSString*)resource
     
     [startUrl appendFormat:@"&%@", [[ADLogger adalId] adURLFormEncode]];
     
-    if (![NSString adIsStringNilOrBlank:userId])
+    if (![NSString adIsStringNilOrBlank:userId.userId])
     {
-        [startUrl appendFormat:@"&%@=%@", OAUTH2_LOGIN_HINT, [userId adUrlFormEncode]];
+        [startUrl appendFormat:@"&%@=%@", OAUTH2_LOGIN_HINT, [userId.userId adUrlFormEncode]];
```

Comment:
UniqueId can be GUID since it is assigned from subjectid/objectid. You can use userid as a loginhint if type is not UniqueId.